engine/hid/src/native/hid_gamepad_driver_glfw.cpp

```diff
@@ -112,6 +125,35 @@ namespace dmHID
         GLFWGamepadDevice new_device = {};
         new_device.m_Id              = gamepad_id;
         new_device.m_Gamepad         = gp;
+        new_device.m_RemapStrategy   = GAMEPAD_REMAP_STRATEGY_NONE;
+
+    #ifdef _WIN32
+        // For windows, we need to be able to remap button layouts according to
+        // how it was created (xinput or dinput) in order to comply with old button mappings.
+
+        new_device.m_RemapStrategy = GAMEPAD_REMAP_STRATEGY_LEGACY_DINPUT;
+
+        char gamepad_names[MAX_GAMEPAD_NAME_COUNT][MAX_GAMEPAD_NAME_LENGTH];
+        uint32_t num_names = GetGamepadDeviceNamesInternal(driver->m_HidContext, gamepad_id, gamepad_names);
+
+        for (int i = 0; i < num_names; ++i)
+        {
+            if (strcmp(gamepad_names[i], "Wireless Xbox Controller") == 0 ||
+                strcmp(gamepad_names[i], "Xbox Controller") == 0 ||
+                strstr(gamepad_names[i], "XInput"))
+            {
+                new_device.m_RemapStrategy = GAMEPAD_REMAP_STRATEGY_LEGACY_XINPUT;
+                break;
+            }

```

Comment:
I don't know how robust this is, but it works for the gamepads I've tested.. This is how they are statically named in GLFW3 now, so the alternative would be to actually patch the GLFW3 source so we could query this directly, but I'd like to avoid that if we can..